commit a5201d42e2f8a8e8062103170027840ee372742f
Author: Schspa Shi <schspa@gmail.com>
Date:   Wed Jun 29 21:09:51 2022 +0800

    regmap: cache: Add extra parameter check in regcache_init
    
    When num_reg_defaults > 0 but reg_defaults is NULL, there will be a
    NULL pointer exception.
    
    Current code has no such usage, but as additional hardening, also
    check this to prevent any chance of crashing.
    
    Signed-off-by: Schspa Shi <schspa@gmail.com>
    Link: https://lore.kernel.org/r/20220629130951.63040-1-schspa@gmail.com
    Signed-off-by: Mark Brown <broonie@kernel.org>

diff --git a/drivers/base/regmap/regcache.c b/drivers/base/regmap/regcache.c
index ef62ec225d81..362e043e26d8 100644
--- a/drivers/base/regmap/regcache.c
+++ b/drivers/base/regmap/regcache.c
@@ -133,6 +133,12 @@ int regcache_init(struct regmap *map, const struct regmap_config *config)
 		return -EINVAL;
 	}
 
+	if (config->num_reg_defaults && !config->reg_defaults) {
+		dev_err(map->dev,
+			"Register defaults number are set without the reg!\n");
+		return -EINVAL;
+	}
+
 	for (i = 0; i < config->num_reg_defaults; i++)
 		if (config->reg_defaults[i].reg % map->reg_stride)
 			return -EINVAL;